erts/emulator/beam/erl_process.c

```diff
@@ -13897,6 +13900,9 @@ erts_continue_exit_process(Process *p)
         /* notify free */
         erts_atomic32_read_bor_relb(&p->state, ERTS_PSFLG_FREE);
 
+        /* clear suspend count */
+        p->rcount = 0;
```

Comment:
Turns out that doing this is not always safe, so need to implement another way to deal with the suspend count.